examples/python/ocr/ocr.py

```diff
@@ -0,0 +1,446 @@
+
+    return rrb.Blueprint(
+        rrb.Vertical(
+            rrb.Horizontal(
+                rrb.Spatial2DView(name="Layout", origin="Image/", contents=["Image/**"] + detections_paths),
+                rrb.Spatial2DView(name="Detections", contents=["Image/**"]),
+                rrb.TextDocumentView(name="Recovery", contents="Recovery"),
+            ),
+            rrb.Horizontal(*tabs),
+            row_shares=[4, 3],
+        ),
+        collapse_panels=True,
+    )
+
+
+def detect_and_log_layout(img_path: str) -> None:
+    # Layout Object - This will contain the detected layouts and their detections
+    layout = Layout()
+
+    # Read Image
+    img = cv2.imread(img_path)
+    coloured_image = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+
+    # Log Image and add Annotation Context
+    rr.log("Image", rr.Image(coloured_image))
+    rr.log(
+        "Image",
+        # The annotation is defined in the Layout class based on its properties
+        rr.AnnotationContext(LayoutType.get_annotation()),
+        timeless=True,
+    )
+
+    # Paddle Model - Getting Predictions
+    logging.info("Start detection... (It usually takes more than 10-20 seconds)")
+    ocr_model_pp = PPStructure(show_log=False, recovery=True)
+    result_pp = ocr_model_pp(coloured_image)
+    _, w, _ = img.shape
+    result_pp = sorted_layout_boxes(result_pp, w)
+    logging.info("Detection finished...")
+
+    # Add results to the layout
+    layout.save_all_layouts(result_pp)
+    logging.info("All results are saved...")
+
+    # Recovery Text Document for the detected text
+    rr.log("Recovery", rr.TextDocument(layout.recovery, media_type=rr.MediaType.MARKDOWN), timeless=True)
+
+    # Generate and send a blueprint based on the detected layouts
+    logging.info("Sending blueprint...")
+    blueprint = generate_blueprint(layout)
+    rr.send_blueprint(blueprint)
+    logging.info("Blueprint sent...")
+
+
+def get_downloaded_path(dataset_dir: Path, video_name: str) -> str:
+    video_file_name = f"{video_name}.mp4"
+    destination_path = dataset_dir / video_file_name
+    if destination_path.exists():
+        logging.info("%s already exists. No need to download", destination_path)
+        return str(destination_path)
+    else:
+        logging.warning("Problem on image downloading")
+        return ""
+
+
+def download_file(url: str, path: Path) -> None:
+    path.parent.mkdir(parents=True, exist_ok=True)
+    logging.info("Downloading %s to %s", url, path)
+    response = requests.get(url, stream=True)
+    with tqdm.tqdm.wrapattr(
+        open(path, "wb"),
+        "write",
+        miniters=1,
+        total=int(response.headers.get("content-length", 0)),
+        desc=f"Downloading {path.name}",
+    ) as f:
+        for chunk in response.iter_content(chunk_size=4096):
+            f.write(chunk)
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="OCR Example - Layout Analysis and Text Detections")
```

Comment:
maybe add a note that this will automatically download the model 🤔 
which btw. in my case took ages since it downloaded it with less than 100kib/s 😞